src/Proto.Actor/HashedConcurrentDictionary.cs

```diff
@@ -32,7 +34,7 @@ internal HashedConcurrentDictionary()
 
         private Dictionary<TKey,TValue> GetPartition(TKey key)
         {
-            var hash = key.GetHashCode() & (0x7FFFFFFF % HashSize);
+            var hash = key.GetHashCode() & HashMask;
```

Comment:
The previous code can't be correct, modulus of this bit mask and then bitwise and of the hashcode.
The new code assumes hash size is a power of two, then creates a mask for all bits below
e.g.
1111111111

This guarantees a positive result, less than HashSize with less computations.